From: Georgia Garcia <georgia.garcia@canonical.com>
Date: Thu, 19 Feb 2026 13:55:02 -0300
Subject: [PATCH 1/3] parser: set umask before creating temp file

Since this file is only used temporarily to load policy, it should be
fine to use S_IRWXU.

Signed-off-by: Georgia Garcia <georgia.garcia@canonical.com>
---
 parser/policy_cache.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/parser/policy_cache.c b/parser/policy_cache.c
index 2b5232e..cc6a9f5 100644
--- a/parser/policy_cache.c
+++ b/parser/policy_cache.c
@@ -220,6 +220,7 @@ int setup_cache_tmp(const char **cachetmpname, char *cachename, int create_compr
 
 	*cachetmpname = NULL;
 	if (write_cache) {
+		mode_t prev_mask;
 		if (create_compressed) {
 			end = strrchr(cachename, '/');
 			*end = '\0';
@@ -236,10 +237,13 @@ int setup_cache_tmp(const char **cachetmpname, char *cachename, int create_compr
 			perror("asprintf");
 			return -1;
 		}
+		prev_mask = umask(S_IRWXU);
 		if ((cache_fd = mkstemp(tmpname)) < 0) {
 			perror("mkstemp");
+			umask(prev_mask);
 			return -1;
 		}
+		umask(prev_mask);
 		*cachetmpname = tmpname;
 	}
 
